flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -293,6 +300,130 @@ class SqlITCase extends StreamingWithStateTestBase {
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+  @Test
+  def testBoundPartitionedEventTimeWindowWithRow(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    env.setStateBackend(getStateBackend)
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.clear
+
+    val t1 = env.fromCollection(data)
+      .assignTimestampsAndWatermarks(new TimestampWithLatenessWatermark())
+      .toTable(tEnv).as('a, 'b, 'c)
+
+    tEnv.registerTable("T1", t1)
+
+    val sqlQuery = "SELECT " +
+      "c, a, " +
+      "sum(a) OVER (PARTITION BY c ORDER BY RowTime() ROWS BETWEEN 2 preceding AND CURRENT ROW)" +
+      "from T1"
+
+    val result = tEnv.sql(sqlQuery).toDataStream[Row]
+    result.addSink(new StreamITCase.StringSink)
+    env.execute()
+
+    val expected = mutable.MutableList(
+      "Hello,1,1", "Hello,2,3", "Hello,3,6", "Hello,4,9", "Hello,5,12",
+      "Hello,6,15", "Hello World,7,7", "Hello World,8,15", "Hello World,20,35")
+    assertEquals(expected.sorted, StreamITCase.testResults.sorted)
+  }
+
+  @Test
+  def testBoundPartitionedEventTimeWindowWithRowWithLateEvent(): Unit = {
+
+    val data = List(
+      (1L, 1, "Hello"),
+      (2L, 2, "Hello"),
+      (4L, 4, "Hello"),
+      (3L, 3, "Hello"),
+      (7L, 7, "Hello"),
+      (7L, 8, "Hello"),
+      (5L, 5, "Hello"),
+      (8L, 8, "Hello World"),
+      (20L, 20, "Hello World"),
+      (9L, 9, "Hello World"))
+
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    env.setStateBackend(getStateBackend)
+    StreamITCase.clear
+
+    // TODO EVENT-TIME OVER window data delay to do a unified configuration,
+    // TODO when we support multi "OVER" window when doing well
+    val config = new JHashMap[String,String]()
+    // need some advice on the definition of this constant
+    config.put(OVER_EVENT_TIME_ALLOWED_LATENESS, "2")
+
+    env.getConfig.setGlobalJobParameters(ParameterTool.fromMap(config))
+
+    // make sure that element (20L, 20, "Hello World") arrived at processing node
+    // followed by element (9L, 9, "Hello World").
+    env.setParallelism(1)
```

Comment:
set the parallelism to 1 such that the test elements are arrived in order. For instance, element (20L, 20, "Hello World") arrives before element (9L, 9, "Hello World").